src/main/java/org/springframework/integration/dsl/config/IntegrationFlowBeanPostProcessor.java

```diff
@@ -143,6 +160,7 @@ private Object processStandardIntegrationFlow(StandardIntegrationFlow flow, Stri
 					endpointBeanName = generateBeanName(endpoint);
 				}
 				registerComponent(endpoint, endpointBeanName, beanName, registerSingleton);
+				integrationComponents.set(i, endpoint);
```

Comment:
ok; but I still think updating the SIF's internal state this way is dirty. If, for some reason, we ever change the getter to return a copy, it won't work.

Perhaps add a new method (`updateComponents()` or similar) ?